projects/kit/src/lib/processors/normalize-date-preprocessor.ts

```diff
@@ -14,53 +14,52 @@ export function normalizeDatePreprocessor({
     dateTimeSeparator?: string;
 }): MaskitoPreprocessor {
     return ({elementState, data}) => {
-        const separator = rangeSeparator
-            ? new RegExp(`${rangeSeparator}|-`)
-            : dateTimeSeparator;
-        const possibleDates = data.split(separator);
-        const dates = data.includes(dateTimeSeparator)
-            ? [possibleDates[0] ?? '']
-            : possibleDates;
+        const dateSegments = data.split(/\D/).filter(Boolean);
+        const templateSegments = dateModeTemplate.split(dateSegmentsSeparator);
+        const includesTime = data.includes(dateTimeSeparator);
 
-        if (
-            dates.every(
-                (date) =>
-                    date.trim().split(/\D/).filter(Boolean).length ===
-                    dateModeTemplate.split(dateSegmentsSeparator).length,
-            )
-        ) {
-            const newData = dates
-                .map((date) =>
-                    normalizeDateString(date, dateModeTemplate, dateSegmentsSeparator),
-                )
-                .join(rangeSeparator);
+        let newData = data;
+        const dates: string[] = [];
+        let dateParts: string[] = [];
+        const timeParts: string[] = [];
 
-            return {
-                elementState,
-                data: `${newData}${
-                    data.includes(dateTimeSeparator)
-                        ? dateTimeSeparator + possibleDates[1] || ''
-                        : ''
-                }`,
-            };
+        for (let index = 0; index < dateSegments.length; index++) {
+            const segment = dateSegments[index]!;
+            const template = templateSegments[index % templateSegments.length];
+            const isLastTemplateSegment =
+                index % templateSegments.length === templateSegments.length - 1;
+
+            if (index >= templateSegments.length && includesTime) {
+                timeParts.push(segment);
+                continue;
```

Comment:
I have an idea how to make code more readable with less variable mutations.

```ts
const dateParts: string[] = segments
    .slice(0, includesTime ? templateSegments.length : Infinity)
    .map((segment) => {
        // ...
    });
const timeParts: string[] = segments
    .slice(includesTime ? templateSegments.length : Infinity)
    .map((segment) => {
        // ...
    });
```

What do you think ? 🤔